### PR TITLE
Expose entity_id in transactions REST API

### LINKED ISSUE #1639: Expose entity_id on /api/v1/transactions endpoint
**Problem**
Currently there's no way to know from an entity impacting transaction what the entity_id is e.g. a transaction that creates a crypto account doesn't say what the entityId is 

**Solution**
Expose the `entity_id` field from the transaction table in the /api/v1/transactions endpoint
```js
{
  ...
  "entity_id":"x.y.z",
  ...
}
```

**Alternatives**

**Additional Context**
The entityId refers to the main entity being impacted.
If there's a secondary entity impacted this would be considered out of scope e.g. TOKEN_ASSOCIATE will refer to the tokenId and not the accountId

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
Currently there's no way to know from an entity impacting transaction what the entity_id is e.g. a transaction that creates a crypto account doesn't say what the entityId is

- Add `entity_id` from transaction into select statement for transactions
- Update spec test files with new column
- Update mock pool file to support new column

**Which issue(s) this PR fixes**:
Fixes #1639 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "transactions": [
       {
         "consensus_timestamp": "1234567890.000000001",
+        "entity_id": null,
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json:40]: Token wipe transaction has an `entity_id` populated with a token ID. Instead of returning null for this and others like it, we should update the  database setup to populate an `entity_id` for transactions that return a non-null `TransactionHandler::getEntity`. This will make the results more accurately represent a production environment. It would also cause less confusion when people look at the test in isolation in the future and wonder why these are null.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json:40]: I see what you're going for which is why I added support to the `addTranaction()` in `integrationDomainOps.js` so transactions would specific the entity_id just like a record would in production.
The challenge with auto populating a non-null value isn the setup vs letting the test logic specify it is that that the transaction itself is the source of the new or existing entity_id that the transaction works on.
You can't be sure what value it is and therefore have to specify it in the transaction input.

File: hedera-mirror-rest/__tests__/integrationDomainOps.js [hunk 2]
```diff
@@ -236,11 +236,11 @@ const addTransaction = async (transaction) => {
 
   const payerAccount = EntityId.fromString(transaction.payerAccountId);
   const nodeAccount = EntityId.fromString(transaction.nodeAccountId, 'nodeAccountId', true);
+  const entityAccount = EntityId.fromString(transaction.entity_id, 'entity_id', true);
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/integrationDomainOps.js:239]: the entity ID of a transaction can be of different type depending on the transaction type, so better name it `entityId`

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json:40]: Yeah, I'm doing this for some tests, you can see the changes in this PR.
Might take a while to go through all the specs to do it everywhere but the capability is added as part of this PR.

I'll work on that

File: hedera-mirror-rest/__tests__/integrationDomainOps.js (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js:239]: good point, will update

File: hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/specs/transactions-30-specific-id-tokenwipe.spec.json]: Updated

File: hedera-mirror-rest/__tests__/integrationDomainOps.js (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-rest/__tests__/integrationDomainOps.js]: Updated

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
